File: 0001-fizz_buzz/ruby/fizz_buzz.rb
#!/usr/bin/env ruby

puts (1..100).map {|i|
  if i % 15 == 0 then 'FizzBuzz'
  elsif i % 3 == 0 then 'Fizz'
  elsif i % 5 == 0 then 'Buzz'
  else i end
}.join(' ')

File: 0001-fizz_buzz/scala/FizzBuzz.scala
object FizzBuzz {
  def main(args: Array[String]) {
    println(
      (1 to 100).map(i =>
        (i % 3, i % 5) match {
          case (0, 0) => "FizzBuzz"
          case (0, _) => "Fizz"
          case (_, 0) => "Buzz"
          case _ => i
        }
      ).mkString(" ")
    )
  }
}
